internal/runners/push/push.go

```diff
@@ -35,76 +34,106 @@ func NewPush(config *viper.Viper, prime primeable) *Push {
-		return failures.FailUserInput.New("err_api_not_authenticated")
+		return locale.NewInputError("err_api_not_authenticated")
 	}
 
 	if r.project != nil && r.project.IsHeadless() {
-		return failures.FailUserInput.Wrap(locale.NewInputError("err_push_headless", "You must first create a project. Please visit {{.V0}} to create your project.", r.project.URL()))
+		return errs.AddTips(locale.NewInputError(
+			"err_push_headless",
+			"You must first create a project."),
+			locale.Tl("push_headless_push_tip_conversion_url", "Visit [ACTIONABLE]{{.V0}}[/RESET] to create a project in the browser.", r.project.URL()),
+			locale.Tl("push_headless_push_tip_state_init", "Run [ACTIONABLE]state init[/RESET] to create a project with the State Tool.."),
+		)
 	}
 
 	// Create the project remotely if it doesn't already exist
 	pjm, fail := model.FetchProjectByName(r.project.Owner(), r.project.Name())
 	if fail != nil {
 		if !fail.Type.Matches(model.FailProjectNotFound) {
-			return fail
+			return locale.WrapError(fail.ToError(), "err_push_try_project", "Failed to check for existence of project.")
 		}
 		// We have to reset handled failures since our legacy command handling still relies on this
-		// ie. failure occurred equals unsuccesful command
+		// ie. failure occurred equals unsuccessful command
 		failures.ResetHandled()
 	}
 
 	if pjm != nil {
 		if r.project.CommitID() == "" {
-			return failures.FailUserInput.New("push_already_exists", r.project.Owner(), r.project.Name())
-		} else {
-			r.Outputer.Notice(locale.T("push_up_to_date"))
+			return locale.NewError("push_already_exists", r.project.Owner(), r.project.Name())
 		}
+		r.Outputer.Notice(locale.T("push_up_to_date"))
 		return nil
 	}
 
-	lang, langVersion, fail := r.languageForProject(r.project)
-	if fail != nil {
-		return fail
+	lang, langVersion, err := r.languageForProject(r.project)
+	if err != nil {
+		return errs.Wrap(err, "Failed to retrieve project language.")
 	}
-
 	r.Outputer.Notice(locale.Tr("push_creating_project", r.project.Owner(), r.project.Name()))
-	var commitID strfmt.UUID
-	pjm, commitID, fail = model.CreateProject(r.project.Owner(), r.project.Name(), model.HostPlatform, lang, langVersion, r.project.Private())
+
+	pjm, fail = model.CreateEmptyProject(r.project.Owner(), r.project.Name(), r.project.Private())
 	if fail != nil {
-		return fail
+		return locale.WrapError(fail.ToError(), "push_project_create_empty_err", "Failed to create an project {{.V0}}.", r.project.Namespace().String())
 	}
+	if r.project.CommitID() != "" {
+		// try to create the project at the given commit id.
+		err := model.UpdateProjectBranchCommit(pjm, r.project.CommitUUID())
+		if err != nil {
+			return locale.WrapError(err, "push_project_branch_commit_err", "Failed to update new project {{.V0}} to current commitID.", r.project.Namespace().String())
+		}
+	} else {
+		_, commitID, fail := model.InitializeProject(r.project.Owner(), r.project.Name(), model.HostPlatform, lang, langVersion)
```

Comment:
You already did this on line 76. What should happen here is creating the commit. The update branch call (currently on line 82) would ALWAYS happen regardless of headless state.

Reply:
I pushed a change to address that.  Note, that `CreateEmptyProject` does not call `InitializeProject` anymore, as you suggested in your comment.